core/src/main/java/org/elasticsearch/index/query/GeoDistanceRangeQueryBuilder.java

```diff
@@ -111,65 +146,295 @@ public GeoDistanceRangeQueryBuilder includeLower(boolean includeLower) {
         return this;
     }
 
+    public boolean includeLower() {
+        return includeLower;
+    }
+
     public GeoDistanceRangeQueryBuilder includeUpper(boolean includeUpper) {
         this.includeUpper = includeUpper;
         return this;
     }
 
+    public boolean includeUpper() {
+        return includeUpper;
+    }
+
     public GeoDistanceRangeQueryBuilder geohash(String geohash) {
         this.geohash = geohash;
         return this;
     }
 
+    public String geohash() {
+        return geohash;
+    }
+
     public GeoDistanceRangeQueryBuilder geoDistance(GeoDistance geoDistance) {
         this.geoDistance = geoDistance;
         return this;
     }
 
+    public GeoDistance geoDistance() {
+        return geoDistance;
+    }
+
+    public GeoDistanceRangeQueryBuilder unit(DistanceUnit unit) {
+        this.unit = unit;
+        return this;
+    }
+
+    public DistanceUnit unit() {
+        return unit;
+    }
+
     public GeoDistanceRangeQueryBuilder optimizeBbox(String optimizeBbox) {
         this.optimizeBbox = optimizeBbox;
         return this;
     }
 
+    public String optimizeBbox() {
+        return optimizeBbox;
+    }
+
     public GeoDistanceRangeQueryBuilder coerce(boolean coerce) {
         this.coerce = coerce;
         return this;
     }
 
+    public boolean coerce() {
+        return coerce;
+    }
+
     public GeoDistanceRangeQueryBuilder ignoreMalformed(boolean ignoreMalformed) {
         this.ignoreMalformed = ignoreMalformed;
         return this;
     }
 
+    public boolean ignoreMalformed() {
+        return ignoreMalformed;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException errors = new QueryValidationException();
```

Comment:
you can use `QueryValidationException#addValidationError` so you don't have to always create the object and return null if empty at the end